src/libraries/System.Private.CoreLib/src/System/TimeOnly.cs

```diff
@@ -0,0 +1,973 @@
+            {
+                if (format[0] == 'o' || format[0] == 'O')
+                {
+                    format = OFormat;
+                    provider = CultureInfo.InvariantCulture.DateTimeFormat;
+                }
+                else if (format[0] == 'r' || format[0] == 'R')
+                {
+                    format = RFormat;
+                    provider = CultureInfo.InvariantCulture.DateTimeFormat;
+                }
+            }
+
+            DateTimeResult result = default;
+
+            result.Init(s);
+
+            if (!DateTimeParse.TryParseExact(s, format, DateTimeFormatInfo.GetInstance(provider), style, ref result))
+            {
+                throw new FormatException(SR.Format(SR.Format_BadTimeOnly, s.ToString()));
+            }
+
+            if ((result.flags & ParseFlagsTimeMask) != 0)
+            {
+                throw new FormatException(SR.Format(SR.Format_TimeOnlyContainsNoneTimeParts, s.ToString()));
+            }
+
+            return new TimeOnly(result.parsedDate.TimeOfDay.Ticks);
+        }
+
+        /// <summary>
+        /// Converts the specified span representation of a time to its TimeOnly equivalent using the specified array of formats.
+        /// The format of the string representation must match at least one of the specified formats exactly or an exception is thrown.
+        /// </summary>
+        /// <param name="s">A span containing the characters that represent a time to convert.</param>
+        /// <param name="formats">An array of allowable formats of s.</param>
+        /// <returns>An object that is equivalent to the time contained in s, as specified by format, provider, and style.</returns>
+        public static TimeOnly ParseExact(ReadOnlySpan<char> s, string [] formats) => ParseExact(s, formats, DateTimeFormatInfo.CurrentInfo, DateTimeStyles.None);
+
+        /// <summary>
+        /// Converts the specified span representation of a time to its TimeOnly equivalent using the specified array of formats, culture-specific format information, and style.
+        /// The format of the string representation must match at least one of the specified formats exactly or an exception is thrown.
+        /// </summary>
+        /// <param name="s">A span containing the characters that represent a time to convert.</param>
+        /// <param name="formats">An array of allowable formats of s.</param>
+        /// <param name="provider">An object that supplies culture-specific formatting information about s.</param>
+        /// <param name="style">A bitwise combination of enumeration values that indicates the permitted format of s. A typical value to specify is None.</param>
+        /// <returns>An object that is equivalent to the time contained in s, as specified by format, provider, and style.</returns>
+        public static TimeOnly ParseExact(ReadOnlySpan<char> s, string [] formats, IFormatProvider? provider, DateTimeStyles style = DateTimeStyles.None)
```

Comment:
I see the typo in many places in some files.
```suggestion
        public static TimeOnly ParseExact(ReadOnlySpan<char> s, string[] formats, IFormatProvider? provider, DateTimeStyles style = DateTimeStyles.None)
```

Reply:
I'll fix those. thanks.